Added Tic Tac Toe Workshop (#1355)
hackclub/hackclub: workshops/tic_tac_toe/README.md

---
name: 'Tic Tac Toe'
description: 'Make the Classic Tic Tac Toe Game in Python'
author: '@Prithul0218'
---

> Tic Tac Toe is a game that most of us have played when we were bored in class. It's also a classic coding challenge when learning a new programming language. As complicated as it may seem at first, it's actually rather easy to code taking less than 80 lines.

Reviewer: ```suggestion
Tic Tac Toe is a game that most of us have played when we were bored in class. It's also a classic coding challenge when learning a new programming language. As complicated as it may seem at first, it's actually rather easy to code takes less than 80 lines.
```

Reviewer 2: Bezlin: this isn't correct grammar it should be "and takes"

> Tic Tac Toe is a game that most of us have played when we were bored in class. It's also a classic coding challenge when learning a new programming language. As complicated as it may seem at first, it's actually rather easy to code taking less than 80 lines.

Reviewer: ```suggestion
Tic Tac Toe is a game that most of us have played when we were bored in class. It's also a classic coding challenge when learning a new programming language. As complicated as it may seem at first, it's actually rather easy to code and takes less than 80 lines.
```

![GIF showing Tic Tac Toe game running in a terminal](https://cloud-hztfs3uyy.vercel.app/ezgif.com-gif-maker.gif)

In this workshop, we will make the Tic Tac Toe game in Python that you can play with a friend using the terminal.

## Getting started

We are going to use [repl.it](http://repl.it) for this project. Just go to [https://repl.it/languages/python3](https://repl.it/languages/python3) to start coding. It's that easy! Creating an account will ensure you don't lose your code but you can do it after we finish this project. Let's start coding!

### Printing the board

> First we need a board to play the game of course. Let's draw on in the terminal! But wait, you can't really draw in the terminal. What we can do though is print horizontal lines with Hyphen `--` and vertical lines with pipes `|`.

Reviewer: ```suggestion
First, we need a board to play the game of course. Let's draw on in the terminal! But wait, you can't really draw in the terminal. What we can do though is print horizontal lines with Hyphen `--` and vertical lines with pipes `|`.
```

Reviewer: Just passed through your workshop found some typos!

    if board[placing_index] == "X" or board[placing_index] == "O":
      print("\nBox already occupied. Try another one")
      Turn(player)
    else:
      board[placing_index] = player

  def CheckWin(player):
    count = 0

    for x in range(9):
      for y in range(9):
        for z in range(9):
          if x != y and y != z and z != x:
            if board[x] == player and board[y] == player and board[z] == player:
              if MagicSquare[x] + MagicSquare[y] + MagicSquare[z] == 15:
                print("Player", player ,"wins!\n")
                return True

    for a in range(9):
      if board[a] == "X" or board[a] == "O":
        count += 1
      if count == 9:
        print("The game ends in a Tie\n")
        return True

  while not end:
    PrintBoard()
    end = CheckWin("O")
    if end == True:
      break
    print("Choose a box player X")
    Turn("X")

    PrintBoard()
    end = CheckWin("X")
    if end == True:
      break
    print("Choose a box player O")
    Turn("O")

TicTacToe()
```

You should now be able to play Tic Tac Toe with a friend in the terminal! You can run my code [here](https://repl.it/@prithul0218/Tic-Tac-Toe).

> ## Hacking

Reviewer 2: @Prithul0218 could you add 3 examples with code? Thanks
